GameServer/Logic/GameObjects/GameObjects.cs

```diff
@@ -19,27 +19,27 @@ public class GameObject : Target
          */
         public Target Target { get; set; }
 
-        protected List<Vector2> waypoints = new List<Vector2>();
-        protected int curWaypoint;
+        public List<Vector2> Waypoints { get; private set; }
+        public int CurWaypoint { get; private set; }
         private TeamId _team;
         public TeamId Team
```

Comment:
This whole property should be refactored a bit. When setting a property does more than just sets the value of it, you should use a function for that.

So essentially this should be
```CS
public TeamId Team { get; private set; }
public void SetTeam(TeamId team)
{
    _visibleByTeam[Team] = false;
    Team = team;
    _visibleByTeam[Team] = true;
}
```